app/server/src/main/scala/org/bitcoins/server/ServerJsonModels.scala

```diff
@@ -483,7 +484,71 @@ object OpReturnCommit extends ServerJsonModels {
             s"Bad number of arguments: ${other.length}. Expected: 3"))
     }
   }
+}
+
+// Oracle Models
+
+case class CreateEvent(label: String, outcomes: Vector[String])
 
+object CreateEvent extends ServerJsonModels {
+
+  def fromJsArr(jsArr: ujson.Arr): Try[CreateEvent] = {
+    jsArr.arr.toList match {
+      case labelJs :: outcomesJs :: Nil =>
+        Try {
+          val label = labelJs.str
+          val outcomes = outcomesJs.arr.map(_.str).toVector
+
+          CreateEvent(label, outcomes)
+        }
+      case Nil =>
+        Failure(
+          new IllegalArgumentException("Missing label and outcome arguments"))
+      case other =>
```

Comment:
i think the only other possibility here is `head :: Nil | head :: tail` isn't it? It's always nice to try and retain pattern match exhaustiveness.

Reply:
Those will have the same error message, so I think it is fine as is. We do this for every other `ServerJsonModel` as well